src/main/java/de/tum/in/www1/artemis/service/connectors/ContinuousIntegrationService.java

```diff
@@ -290,13 +291,16 @@ public String forProgrammingLanguage(ProgrammingLanguage language) {
      * Name of the docker image used for the given programming language.
      *
      * @param language The programming language for which the docker image name is requested
+     * @param projectType The project type of the docker image to return
      * @return The name of the image (published on hub.docker.com)
      */
-    static String getDockerImageName(ProgrammingLanguage language) {
+    static String getDockerImageName(ProgrammingLanguage language, Optional<ProjectType> projectType) {
         return switch (language) {
             case JAVA, KOTLIN, EMPTY -> "ls1tum/artemis-maven-template:java17-2";
             case PYTHON -> "ls1tum/artemis-python-docker:latest";
-            case C -> "ls1tum/artemis-c-docker:latest";
+            case C -> (projectType.isPresent() && projectType.get().equals(ProjectType.FACT))
+                    ? "sharing-codeability.uibk.ac.at:5051/codeability/test-frameworks/fact/fact_artemis:latest"
```

Comment:
Personally, I would say that this is the most critical part :)
I don't know whether this docker registry is available for anyone?

Maybe consider #4455

Reply:
I agree, I cannot access the page. The docker image should be available on DockerHub

Reply:
The docker registry is publicly available and the image is maintained by two instructors at the University of Innsbruck using this Image throughout their courses 🙂 

I think as the main goal of this PR is to share the [actual Framework](https://fact.readthedocs.io/en/latest/) used, an integration in Artemis would better suit the purpose..

Otherwise, local changes are of course much better 🙂

Reply:
The problem with the docker registry is its port :)
Our network at university restricts the outgoing ports (afaik) .
Therefore, I can't test it at our systems:

```
/> curl https://sharing-codeability.uibk.ac.at:5051/codeability/test-frameworks/fact/fact_artemis
curl: (7) Failed to connect to sharing-codeability.uibk.ac.at port 5051: Connection refused
```

At my home systems I can connect to `https://sharing-codeability.uibk.ac.at:5051/codeability/test-frameworks/fact/fact_artemis`

Reply:
@krusche @dfuchss 
An image of FACT is now available on DockerHub, which might solve the access problem. (https://hub.docker.com/r/sharingcodeability/fact)

This has now been replaced with https://github.com/ls1intum/Artemis/pull/4036/commits/9a500ca48f57f7f5716bafbffef9a57b09c7c221

Reply:
Nice work!